var/spack/repos/builtin/packages/chapel/package.py

```diff
@@ -3,24 +3,773 @@
+        multi=False,
+        sticky=True,  # never allow the concretizer to choose this
+    )
+
+    variant(
+        "gasnet_segment",
+        default="unset",
+        description="Build Chapel with multi-locale support using the "
+        "supplied CHPL_GASNET_SEGMENT",
+        values=("everything", "fast", "large", "unset"),
+        multi=False,
+    )
+
+    variant(
+        "gmp",
+        description="Build with gmp support",
+        default="spack",
+        values=("bundled", "none", "spack", "system"),
+        multi=False,
+    )
+
+    variant(
+        "gpu_mem_strategy",
+        description="The memory allocation strategy for GPU data",
+        values=("array_on_device", "unified_memory"),
+        default="array_on_device",
+        multi=False,
+    )
+
+    variant(
+        "host_arch",
+        description="Host architecture of the build machine",
+        values=("x86_64", "aarch64", "arm64", "unset"),
+        default="unset",
+        multi=False,
+    )
+
+    # Feedback that it's hard to imagine any circumstance where host_compiler or
+    # target_compiler could meaningfully differ from what Spack already knows about
+    # variant(
+    #     "host_compiler",
+    #     values=compilers,
+    #     description="Compiler suite for building the Chapel compiler on CHPL_HOST_PLATFORM",
+    #     default="unset",
+    # )
+
+    variant(
+        "host_jemalloc",
+        values=("bundled", "none", "system", "unset"),
+        default="unset",
+        multi=False,
+        description="Selects between no jemalloc, bundled jemalloc, or system jemalloc",
+    )
+
+    variant(
+        "host_mem",
+        values=("cstdlib", "jemalloc"),
+        default="jemalloc",
+        description="Memory management layer for the chpl compiler",
+        multi=False,
+    )
+
+    variant(
+        "host_platform",
+        description="Host platform",
+        default="unset",
+        values=platform_opts,
+        multi=False,
+    )
+
+    variant(
+        "hwloc",
+        description="Build with hwloc support",
+        default="bundled",
+        values=("bundled", "none"),
+        multi=False,
+    )
+
+    variant(
+        "launcher",
+        values=launcher_names,
+        default="unset",
+        description="Launcher to use for running Chapel programs",
+        multi=False,
+    )
+
+    variant(
+        "lib_pic",
+        values=("none", "pic"),
+        default="none",
+        description="Build position-independent code suitable for shared libraries",
+    )
+
+    variant(
+        "libfabric",
+        default="unset",
+        description="When building with ofi support, specify libfabric option",
+        values=("bundled", "system", "unset"),
+        multi=False,
+    )
+
+    variant(
+        "llvm",
+        default="spack",
+        description="LLVM backend type. Use value 'spack' to have spack "
+        "handle the LLVM package",
+        values=("bundled", "none", "spack", "system"),
```

Comment:
so the recommendation is that users would need to have spack detect their system LLVM and provide it as an external package? I think that is fine, this is really a speed/convenience feature to avoid having to compile LLVM for 30+ minutes when the bundled or spack variants are selected on a new system (without an LLVM spackage installed already)

Reply:
> so the recommendation is that users would need to have spack detect their system LLVM and provide it as an external package? 

@arezaii : the `llvm=spack` option already serves two use cases:
1. An appropriate LLVM is available on the system and was detected as an external (e.g. using `spack external find llvm`)
2. LLVM is not built yet, but we want Spack to build it. 
    * This option is notably better than a Chapel-bundled build because it allows one to quickly install multiple `chapel` variants all backed by the same `llvm` build.

So I agree with @scheibelp, we should remove the superfluous `llvm=system` option.

I suspect the same logic implies we should remove `gmp=system` in favor of `gmp=spack`.

We should similarly also consider replacing all of `host_jemalloc=system` and `libfabric=system` and `unwind=system` with `=spack` variants. The Chapel CMake will still treat these "system" installs, but they are all supported by Spack packages that can be either built from source or registered as externals.